### ISSUE TITLE
BUG: Losing levels(regaining professions)

### ISSUE THREAD

**Issue by BurntTrashFor** [Pet-Slime/StardewValley#23]
Due to being able to lose luck levels(from having negative luck), you can go down a level and gain the notification(after you gain back the level) again. This might cause confusion to some people. A bigger problem is the fact that you can lose skill levels after getting a profession, in which you keep the profession but get to choose another one(since you get the level 5/10 level up notification again).
I think this can be fixed by making your Luck skill not being able to go down in level(but still can lose exp).
(The Fortuneteller profession, while make you not go lower than negative luck, also still loses you the original EXP, so if were SUPPOSED to get negative luck, you'll have 0 luck but still lose EXP accordingly)

also, here are some suggestions you can read but dont have to follow:

1. add a config to never lose luck exp from daily luck
2. increase the % of luck you get per Luck Level

That's about it, thank you for your work!

**Comment by Applepatapon** [comment]
I have never coded a stardew mod(except for using PFM but this doesn't count) but I this seems interesting, let me see if I can even attempt to fix this. I might also try to add the suggestions.

**Comment by Applepatapon** [comment]
As stated before, I am not a programmer for stardew mods, and Idk how to use github yet, so unfortunately, from what i see in the code at least, I dont know how to access the current player exp so i couldnt make it. But my possible idea was to first get the current Luck level, then check the minimum(or requirement) to get to that luck level. If you were to reduce it than the minimum, set it to the current minimum.

The first suggestion on the other hand, I think I was able to? imma post my changes(that might or might not work)

in Config.cs:
```
namespace LuckSkill
{
    [SConfig]
    public class Config
    {
        [SConfig.Option(0, 4000, 1)]
        public int DailyLuckExpBonus { get; set; } = 750;
        

>         [SConfig.Option]
>         public bool NegativeEXP = true;

    }
    
}
```
and in Events.cs
```
            if (!Game1.player.IsLocalPlayer)
                return;

            Farmer farmer = Game1.getFarmer(Game1.player.UniqueMultiplayerID);
            int exp = (int)(farmer.team.sharedDailyLuck.Value * ModEntry.Config.DailyLuckExpBonus);

>             if (!ModEntry.Config.NegativeEXP)
>             {
>                 if (exp < 0)
>                         exp = 0;
>             }

            
            Utilities.AddEXP(farmer, exp);
```
Hopefully it helps. If it doesn't I apologize.

**Comment by Pet-Slime** [comment]
Fixed in luck 1.1.0, in which I just made it so you can never gain negative exp.
